README.md

```diff
@@ -6,16 +6,77 @@
+
+#### Clarity
+
+Try to make the `Dockerfile` easy to understand/read. It may be tempting, for the sake of brevity, to put complicated initialization details into a standalone script and merely add a `RUN` command in the `Dockerfile`. However, this causes the resulting `Dockerfile` to be overly opaque, and such `Dockerfile`s are unlikely to pass review. Instead, it it recommended to put all the commands for initialization into the `Dockerfile` as appropriate `RUN` or `ENV` command combinations. To find good examples, look at the current official images.
+
+#### Security
+
+Version bumps and security fixes need to be attended to in a timely manner. Additionally, the `Dockerfile` should be designed to prevent man-in-the-middle attacks: using https where possible; importing PGP keys with the full fingerprint in the Dockerfile to check package signing; embedding checksums directly in the `Dockerfile` if PGP signing is not provided. When importing PGP keys, we recommend using the [high-availability server pool](https://sks-keyservers.net/overview-of-pools.php#pool_ha) from sks-keyservers (`ha.pool.sks-keyservers.net`). Here are a few good and bad examples:
+
+-	**Bad**: *download the file over http with no verification.*
+	```Dockerfile
+	RUN curl -fSL "http://julialang.s3.amazonaws.com/bin/linux/x64/${JULIA_VERSION%[.-]*}/julia-${JULIA_VERSION}-linux-x86_64.tar.gz" | tar ... \
+		# install
+	```
+
+-	**Good**: *download the file over https, but still no verification.*
+	```Dockerfile
+	RUN curl -fSL "https://julialang.s3.amazonaws.com/bin/linux/x64/${JULIA_VERSION%[.-]*}/julia-${JULIA_VERSION}-linux-x86_64.tar.gz" | tar ... \
+		# install
+	```
+
+-	**Better**: *embed the checksum into the Dockerfile. It would be better to use https here too, if it is available.*
+	```Dockerfile
+	ENV RUBY_DOWNLOAD_SHA256 5ffc0f317e429e6b29d4a98ac521c3ce65481bfd22a8cf845fa02a7b113d9b44
+	RUN curl -fSL -o ruby.tar.gz "http://cache.ruby-lang.org/pub/ruby/$RUBY_MAJOR/ruby-$RUBY_VERSION.tar.gz" \
+		&& echo "$RUBY_DOWNLOAD_SHA256 *ruby.tar.gz" | sha256sum -c - \
+		# install
+	```
+
+-	**Best**: *full key fingerprint imported to apt-key which will check signatures when packages are downloaded and installed.*
+	``` Dockerfile
+	RUN apt-key adv --keyserver ha.pool.sks-keyservers.net --recv-keys 492EAFE8CD016A07919F1D2B9ECBEC467F0CEB10
+	RUN echo "deb http://repo.mongodb.org/apt/debian wheezy/mongodb-org/$MONGO_MAJOR main" > /etc/apt/sources.list.d/mongodb-org.list
+	RUN apt-get update \
+		&& apt-get install -y mongodb-org=$MONGO_VERSION \
+		&& rm -rf /var/lib/apt/lists/* \
```

Comment:
What's this `rm -rf /var/lib/apt/lists/*` business and why should I use it??? :trollface: